astroid/rebuilder.py

```diff
@@ -2125,11 +2125,20 @@ def visit_starred(self, node: "ast.Starred", parent: NodeNG) -> nodes.Starred:
     def visit_tryexcept(self, node: "ast.Try", parent: NodeNG) -> nodes.TryExcept:
         """visit a TryExcept node by returning a fresh instance of it"""
         if sys.version_info >= (3, 8):
+            # TryExcept excludes the 'finally' but that will be included in the
+            # end_lineno from 'node'. Therefore, we iterate over all non 'finally'
+            # children to find the correct end_lineno and column.
+            last_child_end_lineno = node.lineno
+            last_child_end_col_offset = node.end_col_offset
+            for child in node.body + node.handlers + node.orelse:
+                if child.end_lineno > last_child_end_lineno:
+                    last_child_end_lineno = child.end_lineno
+                    last_child_end_col_offset = child.end_col_offset
```

Comment:
Can't we just always use the last child of `node.body + node.handlers + node.orelse`. The `try` statement has a pretty strict structure, i.e. the if there is an `else` block, it's always **after** the `except` ones.

Reply:
Not sure what you meant. See the latest commit, is that what you were looking for?

Reply:
Not exactly. I had something like this in mind (if it works)
```py
            end_lineno = node.end_lineno
            end_col_offset = node.end_col_offset
            all_children: List["ast.AST"] = [*node.body, *node.handlers, *node.orelse]
            for child in reversed(all_children):
                end_lineno = child.end_lineno
                end_col_offset = child.end_col_offset
                break
```